src/app/Analytics/LocalMonitoring/RecommendationsForLocalMonitoring/RemoteMonitoring/RecommendationTables.tsx

```diff
@@ -96,11 +109,67 @@ const RecommendationTables = (props: { endTimeArray; setEndTimeArray; SREdata; s
         setCurrentData(current_arr);
         setRecommendedData(recommended_arr);
         setChartDetailsData(chartDetailsObject);
-      }
-    };
+          // setBoxPlotData(boxPlot_arr);
 
-    fetchData();
-  }, [endtime, day, props.SREdata.experiment_name]);
+          const chartData = Object.keys(boxPlot_arr).map((key) => {
+            if (boxPlot_arr && boxPlot_arr[0]) {
+              const timestamps = Object.keys(boxPlot_arr[0]);
+              const cpuDetails = Object.values(boxPlot_arr[0]).map((key) => key.cpuUsage);
+  
+              return {
+                name: 'cpu',
+                x: timestamps,
+                y: cpuDetails
+              };
+            } else return 0;
+          });
+  
+          // console.log(chartData);
+          const transformedData =
+            chartData &&
+            chartData.map((containerData) => {
+              const timestamps = containerData.x;
+              const yData = containerData.y.map((data) => [data.min, data.q1, data.median, data.q3, data.max]);
+  
+              return {
+                name: 'cpu',
+                x: timestamps,
+                y: yData
+              };
+            });
+  
+            // for box plots data points
```

Comment:
follow the same pattern to comment or describe the section of code.
No extra line between comments and code.

Reply:
done